Data/DataModel/HelpMessage.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace ServerCore.DataModel
+{
+    /// <summary>
+    /// Message in help thread when a player asks for help.
+    /// </summary>
+    public class HelpMessage
+    {
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        public int ID { get; set; }
+
+        /// <summary>
+        /// The date time (in UTC) of when the message was sent.
+        /// </summary>
+        public DateTime DateTimeInUtc { get; set; }
+
+        /// <summary>
+        /// The puzzle id that this message is for.
+        /// </summary>
+        [Required]
+        public int PuzzleID { get; set; }
+
+        /// <summary>
+        /// The puzzle this is a help message for.
+        /// </summary>
+        [Required]
+        public virtual Puzzle Puzzle { get; set; }
+
+        /// <summary>
+        /// The actual message text.
+        /// </summary>
+        public string Text { get; set; }
+
+        /// <summary>
+        /// True if message was from the player (as opposed to a response from puzzle control).
+        /// </summary>
+        public bool IsPlayerMessage { get; set; }
+
+        /// <summary>
+        /// The team ID of the sender (will be empty if not a player of if it is for a SinglePlayerPuzzle).
```

Comment:
If it could be empty then the value should be nullable (i.e. int?, not int).

Also, I think that (puzzle x team) or (puzzle x individual) are required scopes for a puzzle message, so I think calling these the "sender" is too limiting. For example, when puzzle authors respond to a message on a team puzzle, the team ID needs to be filled out.

Reply:
I made a PuzzleMessage, which has a Puzzle, which will know if it is a singlePlayerPuzzle or not.
When looking at PuzzleMessages, we could join with team members to find messages viewable to the entire team.
This is so that we can put GameControl replies and player messages in the same database.

Or do you think it would be worth it to put them into three separate databases (PlayerMessage, TeamMessage, and GameControlMessages)?

Reply:
Actually, I changed my mind and made a teamMessage